File: python/src/search/linear_search.py
from typing import List, Any, Optional

def linear_search(data: List[Any], target: Any) -> int:
    """
    線形探索を用いてリストからターゲットを探します。
    
    Args:
        data (List[Any]): 探索対象のリスト
        target (Any): 探索する値
        
    Returns:
        int: 見つかった場合はそのインデックス、見つからない場合は -1
    """
    for i, value in enumerate(data):
        if value == target:
            return i
    return -1

File: javascript/src/search/linear_search.js
/**
 * 線形探索を用いて配列からターゲットを探します。
 * 
 * @param {Array} array 探索対象の配列
 * @param {*} target 探索する値
 * @returns {number} 見つかった場合はそのインデックス、見つからない場合は -1
 */
function linearSearch(array, target) {
    for (let i = 0; i < array.length; i++) {
        if (array[i] === target) {
            return i;
        }
    }
    return -1;
}

module.exports = { linearSearch };
